rex/renewable_resource.py

```diff
@@ -102,7 +107,7 @@ def _preload_SAM(self, sites, tech='pvwattsv7', clearsky=False,
         SAM_res.check_irradiance_datasets(self.datasets, clearsky=clearsky)
         for var in SAM_res.var_list:
             if var in self.datasets:
-                SAM_res[var] = self[var, :, sites]
+                SAM_res[var] = self[var, time_slice, sites]
```

Comment:
This will still have to read all of the data from the resource file right? Have you ever tested if reading something like `slice(None, None, 2)` is slower than reading the full data and then slicing it in-memory? I'm wondering if h5py tries to do anything fancy behind the scenes.

Reply:
That is a complicated question. Depending on the step size you could end up reducing the amount of data being read from disk, but it would have to be a really big step-size. That said, h5py and HSDS both have an internal cache so if you make the same data request multiple times it pulls it from the cache instead of from disk. Regardless h5py and hsds will both only return what we request to memory so we should still see performance improvements from that end.

Reply:
okay sounds good! Just want to keep picking your brain on how h5py works on a low level :)